client/src/components/Patients/index.js

```diff
@@ -1,32 +1,64 @@
 import React, { useState, useEffect } from 'react';
-import { Table, Typography, Input, message, Spin } from 'antd';
+import { Table, Typography, Input, message, Spin, Select } from 'antd';
 import axios from 'axios';
 import './style.css';
 import { useHistory } from 'react-router-dom';
+import { differenceInCalendarYears, parse } from 'date-fns';
 
 const { Search } = Input;
 const { Title } = Typography;
+const { Option } = Select;
 
 const Patients = () => {
   const [dataSource, setDataSource] = useState([]);
   const [loading, setLoading] = useState(true);
+  const [searchKey, setSearchKey] = useState('firstName');
   const history = useHistory();
 
-  const fetchPatients = async () => {
+  const buildUrlQuery = (params) => {
+    const usp = new URLSearchParams();
+    const newParams = params || {};
+    Object.keys(newParams).forEach((key) => {
+      const value = newParams[key];
+      if (value) usp.set(key, value);
+    });
+    return usp.toString();
+  };
+
+  const fetchPatients = async (searchQuery) => {
     try {
+      setLoading(true);
+      if (typeof searchQuery === 'string' && searchQuery.trim().length) {
+        setLoading(true);
+        const search = {};
+        search[searchKey] = searchQuery;
+        const {
+          data: { data },
+        } = await axios.get(`/api/v1/patients/search?${buildUrlQuery(search)}`);
+        setDataSource(data);
+        setLoading(false);
+        return data;
+      }
       setLoading(true);
       const {
         data: { data },
       } = await axios.get(`/api/v1/patients`);
```

Comment:
I think you don't have to get all the data, the patient component doing this when it starts, so you can only re-trigger the useEffect of that component and you will get all the data.